overlord/hookstate/hookmgr.go

```diff
@@ -113,6 +125,12 @@ func (m *HookManager) Stop() {
 	m.runner.Stop()
 }
 
+// doRunTool is used to run the requested tool.
+func (m *HookManager) doRunTool(request ToolRequest) (stdout string, stderr string) {
```

Comment:
Ah, and I think we want an error out of it, when we couldn't even run the action for whatever reason (non-existing context, etc).